apps/andark/app.js

```diff
@@ -117,36 +128,49 @@ function drawScale(){
     g.fillRect(10,10,2*c.x-10,2*c.x-10);
     g.setColor(1,1,1);
   }
-}
+};
 
 //// main running sequence ////
 
 // Show launcher when middle button pressed, and widgets that we're clock
-Bangle.setUI("clock");
+Bangle.setUI({
+  mode: "clock",
+  remove: function() {
+    if (drawTimeout) clearTimeout(drawTimeout);
+    drawTimeout = undefined;
+    Bangle.removeListener('lcdPower', updateState);
+    Bangle.removeListener('lock', updateState);
+    require("widget_utils").show();
+}});
 // Load widgets if needed, and make them show swipeable
 if (settings.loadWidgets) {
   Bangle.loadWidgets();
   require("widget_utils").swipeOn();
 } else if (global.WIDGETS) require("widget_utils").hide();
-// Clear the screen once, at startup
-drawScale();
-draw();
 
-let secondInterval = setInterval(draw, 1000);
+let updateState = function() {
+   if (Bangle.isLCDOn()) {
+     if (!Bangle.isLocked()) {
+       queueMillis = 1000;
+       unlock = true;
```

Comment:
Is there any performance penalty (or more like power usage) calling `Bangle.isLocked()` on every draw? Otherwise, I'm fine with the change for cleaner code

Reply:
I suspect so, probably quite a few more instructions - let's keep it as-is in that case